### PR TITLE
Add support for masked components

### PR DESCRIPTION
**Pull request by daniel-thom**
This PR enables full support of HybridSystems in PowerSystems. PowerSystems code will "mask" all subcomponents in a HybridSystem. Those subcomponents will still be managed by the system but will not be exposed in functions like `get_components`.

### CODE REVIEW

File: src/system_data.jl [hunk 1]
```diff
@@ -314,12 +327,33 @@ function remove_components!(::Type{T}, data::SystemData) where {T}
...
 function clear_time_series!(data::SystemData)
     clear_time_series!(data.time_series_storage)
     clear_time_series!(data.components)
     reset_info!(data.time_series_params)
 end
 
+function iterate_components_with_time_series(data::SystemData)
+    return Iterators.Flatten((
+        iterate_components_with_time_series(data.components),
+        iterate_components_with_time_series(data.masked_components),
```
Comment (daniel-thom) [inline src/system_data.jl:353]: Note that we need to return time series of masked components in order to support `transform_single_time_series!` and `get_time_series_multiple`.

File: src/components.jl [hunk 2]
```diff
@@ -50,7 +51,9 @@ function add_component!(
 
     !skip_validation && check_component(components, component)
 
-    if !deserialization_in_progress && has_time_series(component)
+    if !deserialization_in_progress &&
```
Comment (jd-lara) [inline src/components.jl:54]: can we make an intermediate var here to make it easier to read.

**Review by jd-lara — APPROVED** [review]

File: src/components.jl (hunk 2 above)
Comment (daniel-thom) [reply src/components.jl:54]: Can you be more specific? I'm not following.

Comment (jd-lara) [reply src/components.jl:54]:
```suggestion
    time_series_condition = !allow_existing_time_series && has_time_series(component)
    if !deserialization_in_progress && time_series_condition
```

Comment (jd-lara) [reply src/components.jl:54]: Its a minor thing just to understand what the condition is about

Comment (daniel-thom) [reply src/components.jl]: I took the point that it wasn't easy to read and simplified the function.

File: src/components.jl [hunk 3]
```diff
@@ -35,7 +35,7 @@ function add_component!(
     components::Components,
     component::T;
     skip_validation = false,
-    deserialization_in_progress = false,
+    allow_existing_time_series = false,
```
Comment (jd-lara) [inline src/components.jl:38]: @daniel-thom This broke the API... we needed to add a deprecation

Comment (daniel-thom) [reply src/components.jl:38]: As far as I know, PowerSystems is the only module that relies on this, so I think we're safe changing it. It does not affect `PowerSystems.add_components`.

**Pull request merged**